cmake/CeleritasMakeRulesOverride.cmake

```diff
@@ -1,40 +0,0 @@
-#------------------------------- -*- cmake -*- -------------------------------#
-# Copyright Celeritas contributors: see top-level COPYRIGHT file for details
-# SPDX-License-Identifier: (Apache-2.0 OR MIT)
-#[=======================================================================[.rst:
-
-CeleritasMakeRulesOverride
---------------------------
-
-Override CMake platform information with defaults. This is the cleanest way to
-change default flags in a safe (platform-independent and non-user-ignoring) way.
-See
-https://stackoverflow.com/questions/28732209/change-default-value-of-cmake-cxx-flags-debug-and-friends-in-cmake
-and
-https://cmake.org/cmake/help/latest/variable/CMAKE_USER_MAKE_RULES_OVERRIDE.html
-
-#]=======================================================================]
-
-# Enable lots of warnings for GCC and Clang by default
-foreach(_lang C CXX)
-  set(_id "${CMAKE_${_lang}_COMPILER_ID}")
-  if(_id STREQUAL "GNU" OR _id MATCHES "Clang$")
-    string(APPEND CMAKE_${_lang}_FLAGS_INIT "-Wall -Wextra -pedantic")
```

Comment:
Should we mention the dropping (no longer the default for Celeritas build) of those options? (Even if they are in all the presets, they would be 'gone' from a non-preset build).

Reply:
If it does show in the release note, that would be sufficient I think.